lib/tasks/new_adapter.rake

```diff
@@ -0,0 +1,57 @@
+def extract_adapter_template_code(file_name, template_code_str)
```

Comment:
It is not good to have one single method that do all the things. We should have method to generate string for internal methods like `init`, `export_html_file`.

There are many ways to make it modular & extendable. Think/design good one.